docs/content/en/highlights/2021-07-13-helm-customConfig.md

```diff
@@ -0,0 +1,324 @@
+---
+date: "2021-07-13"
+title: "Introduction of `customConfig` and deprecation notice for TOML based config keys"
+short: "Introduction of `customConfig`"
+description: "Configure Vector directly through your `values.yaml` without having to converting to TOML"
+authors: ["spencergilbert"]
+pr_numbers: [8079]
+release: "0.15.0"
+hide_on_release_notes: false
+badges:
+  type: "deprecation"
+  platforms: ["helm"]
+  domains: ["config"]
+---
+
+With the release of Vector 0.15.0, we have introduced a new method of configuring Vector through Helm.
+This new method uses YAML configuration files, and in a coming release will become the default configuration
+method.
+
+**You do not need to upgrade immediately. The deprecated keys will not be removed before Vector hits 1.0**
+
+## Upgrade Guide
+
+### Breaking: updated container args
+
+The argument passed to Vector has been changed to `--config-dir` which will load any TOML, YAML, or JSON
+files added to the "/etc/vector" directory.
+
+### globalOptions and logSchema
+
+Any [global options](https://vector.dev/docs/reference/configuration/global-options/) can be moved directly
+into the `customConfig` key and converted into snake-case.
+
+```diff title="values.yaml"
+   globalOptions:
++    enabled: false
+     dataDir: "/vector-data-dir"
+   logSchema:
++    enabled: false
+     hostKey: "host"
+     messageKey: "message"
+     sourceTypeKey: "source_type"
+     timestampKey: "timestamp"
+   ...
++  customConfig:
++    data_dir: "/vector-data-dir"
++    log_schema:
++      host_key: host
++      message_key: message
++      source_type_key: source_type
++      timestamp_key: timestamp
++  ...
+```
+
+### vectorApi
+
+[Vector API](https://vector.dev/docs/reference/api/) options can be moved as is under a `customConfig.api` key.
+The `extraContainerPorts` or `service` key should be used to expose the port configured in `customConfig`.
+
+```diff title="values.yaml"
+   vectorApi:
++    enabled: false
+     address: "0.0.0.0:8686"
+     playground: true
+   ...
++  customConfig:
++    ...
++    api:
++      enabled: true
++      address: 0.0.0.0:8686
++      playground: true
++    ...
+```
+
+### kubernetesLogsSource
+
+The vector-agent chart will continue to mount the hostPaths required to access Pod logs.
+
+```diff title="values.yaml"
+   kubernetesLogsSource:
++    enabled: false
+     sourceId: kubernetes_logs
+     config: {}
+     rawConfig: null
```

Comment:
This seems strange. If they're adding the source in the `customConfig`, shouldn't we instruct them to remove this entire block instead?

Reply:
I think until we default these off, or remove them entirely for defaults, users will need to provide a `<component>.enabled: false` for things to work properly